tools/upload_text.php

```diff
@@ -269,12 +269,55 @@
         // replace filename
         $zipext = ".zip";
         $name = $projectid.$indicator.$zipext;
-        $path_to_file = "$projects_dir/$projectid/";
-        $location = $path_to_file.$name;
+        $location = "$project->dir/$name";
         ensure_path_is_unused( $location );
         copy($uploaded_file, $location);
-        unlink($uploaded_file);
         $have_file = TRUE;
+        if ($stage == 'smooth_avail')
+        {
+            $project->delete_smoothreading_dir();
+            $smooth_dir = "$project->dir/smooth";
+            if(!mkdir($smooth_dir))
+            {
+                die("Could not create smooth directory");
+            }
+            // unzip into smooth folder
+            $zip = new ZipArchive;
```

Comment:
Consider using `extract_zip_to()` in `misc.inc`.

Reply:
Yes, this should work.